paddle/fluid/framework/parallel_executor.cc

```diff
@@ -290,6 +290,14 @@ ParallelExecutor::ParallelExecutor(const std::vector<platform::Place> &places,
   member_->use_all_reduce_ =
       build_strategy.reduce_ == BuildStrategy::ReduceStrategy::kAllReduce;
   member_->nranks_ = build_strategy.num_trainers_ * places.size();
+#if defined(PADDLE_WITH_CUDA) && defined(_WIN32)
+  if (!member_->use_all_reduce_) {
+    LOG_FIRST_N(WARNING, 1)
+        << "Please NOTE: Windows can support single GPU only"
+        << ", parallelExecutor would be forced to use allreduce.";
+    member_->use_all_reduce_ = true;
+  }
+#endif
```

Comment:
Why do you add those?
Currently, it is not supported that using multi-GPU to train or test, so you should check the number of places first.

Reply:
ok